### PR TITLE
Fix loading data from Excel file into DataHarmonizer 

### LINKED ISSUE #993: Import XLSX file sends incremental record save request but doesn't update interface
1. From the `/samples` page of a new submission, click the Download XLSX button
2. In Excel fill in a few values for the first column and save (for example: [nmdc_sample_export.xlsx](https://github.com/microbiomedata/nmdc-server/files/12073288/nmdc_sample_export.xlsx))
3. Use the Import XLSX File button to load the updated Excel file

Expected behavior: data from the Excel sheet is displayed in the interfaced and sync'd to the server
Actual behavior: data is only sync'd to server (i.e. it appears if the `/samples` page is refreshed)

This might be fallout from the fix for #916

### PR DESCRIPTION
**Pull request by pkalita-lbl**
Fixes #993 -- see issue for reproduction steps.

These changes ensure that data and validation state are fully updated while importing an Excel file.

### CODE REVIEW

**Comment by pkalita-lbl** [comment]
@naglepuff I merged main. Unclear why the server workflow is failing. These are frontend-only changes.

**Review by naglepuff — APPROVED** [review]
LGTM. Good catch. Not sure why the tests are failing, but it seems to have something to do with some issues with the build :/

I'll leave an approval here but I need to investigate the test issues.

**Comment by pkalita-lbl** [comment]
Okay, I'm not authorized to merge without required checks passing. So I'll leave it up to you to override and merge, or I'll wait for a fix to go into `main` and merge this branch again.

**Pull request merged**
